### ISSUE TITLE
Nullary enums:  unresolved name 'Foo'

### ISSUE THREAD

**Issue by steveklabnik** [rust-lang/rust#29652]
Working on some stuff with @petehunt this evening, we found a really bad diagnostic. Pete is getting started with Rust, and had some code that boils down to this:

```
enum Foo {}

fn foo() {
    Foo.bar();
}
```

This gives a really, really poor error:

```
src/lib.rs:4:5: 4:8 error: unresolved name `Foo` [E0425]
src/lib.rs:4     Foo.bar();
                 ^~~
```

It took me a few minutes to diagnose what was going on. If `Foo` has any variants, the error is way better. Code:

```
enum Foo {
    Bar
}

fn foo() {
    Foo::Bar.bar();
}
```

error:

```
src/lib.rs:6:14: 6:19 error: no method named `bar` found for type `Foo` in the current scope
src/lib.rs:6     Foo::Bar.bar();
                          ^~~~~
```

It'd be nice if the nullary enum had the same error message.

**Comment by apasel422** [comment]
Should this actually refer to nullary enums?

**Comment by steveklabnik** [comment]
yes, thanks. it's late here :sweat:

**Comment by eddyb** [comment]
But the "better" error message is for a completely different thing.
`Foo` doesn't name a value, only a type, in `enum Foo {}`.

**Comment by arielb1** [comment]
@steveklabnik 

You will still get this error if you refer to an enum (or even a dict-like struct) with variants:
```Rust
enum Foo { Bar }

fn main() {
    Foo.bar(); //~ ERROR unresolved name `Foo`
}
```

**Comment by steveklabnik** [comment]
@arielb1 ah, nice catch.

So, while I see your point, @eddyb , the issue here is that "unresolved name" makes me think that something isn't in scope that should be. I spent a lot of time re-checking paths, Cargo.toml, and the source code of the dependent library (not needed to reproduce the error, but the enum was external before).

Second,

>  Foo doesn't name a value, only a type, in enum Foo {}.

The second error message says nothing about values, it says type.


Anyway, what I'm saying is that "unresolved name" is confusing, as the name _is_ resolved. I'm not totally sure how to succinctly describe what it should be right this second, though. I like the other error message because it makes me think "why isn't that method defined? Oh, because I used the wrong syntax." not "I need to double check all my use statements."
